packages/mui-system/src/Grid/createGrid.tsx

```diff
@@ -169,7 +169,12 @@ export default function createGrid(
         {...other}
       >
         {React.Children.map(children, (child) => {
-          if (React.isValidElement(child) && isMuiElement(child, ['Grid'])) {
+          if (
+            React.isValidElement(child) &&
+            isMuiElement(child, ['Grid']) &&
+            container &&
```

Comment:
I don't think checking for `container` is correct: https://github.com/mui/material-ui/blob/be183c4d560a8343f82c284dd407f618b54f06a2/packages/mui-system/src/Grid/Grid.tsx#L111C6-L128C29. The level should be increased regardless.

Reply:
The actual level number has no meaning anymore after this change. The only way it now is being used is to check whether it's 0 or greater. It's now only used to find out whether the current grid is nested (as per what the docs define as nested, container inside of container) or top-level. I believe we need to update that comment you refer to instead 🤔

Reply:
Oh, ok, that makes sense. Yes, if the logic is changed, let's update the comment.